lib/parsers/json.js

```diff
@@ -775,6 +772,13 @@ class JoiJsonSchemaParser {
       delete schema.type
     }
   }
+
+  _moveSchemaToDefinitions(schema, joiSpec, definitions, schemaId) {
+    definitions[schemaId] = schema
+    return {
+      $ref: `${this._getLocalSchemaBasePath()}/${schemaId}`
+    }
+  }
```

Comment:
Preserve current top-level behaviour

Reply:
I think it might not need to preserve the top-level behavior when there is "id" for the schema as it can be referenced anyway.

Reply:
Right, what I meant was that I'll just leave this base json converter unchanged since it also has certain copy behaviour and does not seem to support any metas. Does that make sense or did I miss something?